src/Path/Renamer.hs

```diff
@@ -16,24 +18,26 @@ import Path.Term
 import Text.PrettyPrint.ANSI.Leijen hiding ((<$>))
 import Text.Trifecta.Rendering (Span)
 
-resolveTerm :: (Carrier sig m, Member (Error ResolveError) sig, Member (Reader ModuleName) sig, Member (Reader Resolution) sig, Monad m)
-            => Term (Surface Name) Span
-            -> m (Term (Surface QName) Span)
+resolveTerm :: (Carrier sig m, Member (Error ResolveError) sig, Member Fresh sig, Member (Reader ModuleName) sig, Member (Reader Resolution) sig, Monad m)
+            => Term (Surface (Maybe Name) Name) Span
+            -> m (Term (Surface Name QName) Span)
```

Comment:
The renamer now eliminates `Maybe`s in local binding positions, constructing `Gensym` names as appropriate.